ui/components/app/transaction-detail/transaction-detail.component.js

```diff
@@ -2,11 +2,94 @@ import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 
 import { I18nContext } from '../../../contexts/i18n';
+import { useGasFeeContext } from '../../../contexts/gasFee';
+import InfoTooltip from '../../ui/info-tooltip/info-tooltip';
+import Typography from '../../ui/typography/typography';
 
 import TransactionDetailItem from '../transaction-detail-item/transaction-detail-item.component';
+import { COLORS } from '../../../helpers/constants/design-system';
+
+const GasLevelIconMap = {
+  low: '🐢',
+  medium: '🦊',
+  high: '🦍',
+  dappSuggested: '🌐',
+  custom: '⚙',
+};
 
 export default function TransactionDetail({ rows = [], onEdit }) {
+  // eslint-disable-next-line prefer-destructuring
+  const EIP_1559_V2 = process.env.EIP_1559_V2;
+
   const t = useContext(I18nContext);
+  const {
+    estimateToUse,
+    gasLimit,
+    gasPrice,
+    isUsingDappSuggestedGasFees,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    transaction,
+    supportsEIP1559,
+  } = useGasFeeContext();
+  const estimateUsed = isUsingDappSuggestedGasFees
+    ? 'dappSuggested'
+    : estimateToUse;
+
+  if (EIP_1559_V2 && estimateUsed) {
+    return (
+      <div className="transaction-detail">
+        <div className="transaction-detail-edit-V2">
+          <button onClick={onEdit}>
+            <span className="transaction-detail-edit-V2-icon">
+              {`${GasLevelIconMap[estimateUsed]} `}
+            </span>
+            <span className="transaction-detail-edit-V2-label">
+              {t(estimateUsed)}
+            </span>
+            <i className="fas fa-chevron-right asset-list-item__chevron-right" />
+          </button>
+          {estimateUsed === 'custom' && onEdit && (
+            <button onClick={onEdit}>{t('edit')}</button>
+          )}
+          {estimateUsed === 'dappSuggested' && (
+            <InfoTooltip
+              contentText={
+                <div className="transaction-detail-edit-V2-tooltip">
+                  <Typography fontSize="12px" color={COLORS.GRAY}>
```

Comment:
The color grey is currently available under `COLORS.GREY`, `COLORS.GRAY` is no longer defined, see https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/constants/design-system.js#L13